metadetect/masking.py

```diff
@@ -3,6 +3,55 @@
 from .interpolate import interpolate_image_at_mask
 
 
+def apply_apodization_corrections(*, mbobs, ap_rad, mask_bit_val):
+    """Apply an apodization mask around the edge of the images in an mbobs to
+    prevent FFT artifacts.
+
+    Parameters
+    ----------
+    mbobs: ngmix.MultiBandObsList
+        The observations to mask
+    mask_bit_val: int
+        The bit to set in the bit mask for areas that are apodized.
+    ap_rad: float
+        When apodizing, the scale of the kernel. The total kernel goes from 0 to 1
+        over 6*ap_rad.
+    """
+    ap_mask = np.ones_like(mbobs[0][0].image)
+    _build_square_apodization_mask(ap_rad, ap_mask)
+
+    msk = ap_mask < 1
```

Comment:
Using a bool mask `ap_mask < ` means you need to check it at every pixel, every time it is used.  when you do `image[msk]` it has to check "is msk True?" at every pixel.  In this case a small fraction of the pixels are going to have `ap_mask < 1` so its faster to do
```python
msk = np.where(ap_mask < 1)
if msk[0].size > 0:
    for obslist in mbobs
        ...
        obs.image[msk] *= ap_mask[msk]
```
later you do `np.all(msk)`, again hitting every pixel, but with above you can again check `msk[0].size == image.size`.

Maybe this optimization doesn't matter, maybe it does.

Reply:
Good points! Fixed!